### PR TITLE
Refactor citations

### PR DESCRIPTION
**Pull request by vincerubinetti**
Overall purpose of this PR: clean up, consistency, and ability to support in-order citation numbering  

- tweak "show more lines" css styles
- rename "data" and "datum"/"d" to "loci" and "locus" to be more descriptive and consistent with filenemaes
- move all processing of locus data to derived.js, like finding of same genes, extracting in-text references to citations, filling in citation details, and assigning citation numbers
- number citations in same place (derived.js), in order that they appear in the specified and ordered fields, and subsequently sort references field in same order
- tweak hash glow animation
- change hash ids of citations to use the actual ids. that is, if you click on a number reference in e.g. the details field, the url will go to e.g. "#mondo:12345" instead of "#citation-1".
- move in-text citation extraction logic from Cited component to derived.js as aforementioned

### CODE REVIEW

File: site/src/data/derived.js [hunk 1]
```diff
@@ -1,34 +1,129 @@
...
+  }
+
+  if (citations) {
+    /** map for quick lookup of citation by id */
+    const citationLookup = Object.fromEntries(
+      citations.map((citation) => [citation.id, citation]),
+    );
+
+    /** fields that may contain in-text citations, in order they appear on locus page (so number increases as you go down page) */
+    const inTextFields = [
```
Comment (vincerubinetti) [inline site/src/data/derived.js:48]: Here is where we specify which fields, and in what order, to extract (and thus number) citations. Remember to update this if you re-order these fields on the page, or if you add a new field that has in-text references.

**Comment by vincerubinetti** [comment]
I just noticed this on `eiee1_arx`:

Screenshot 2024-12-04 at 12 28 06 AM

It's missing the number. Looking at the locus data, I see it has `"year": "2002 [@pmid:11889467]"`. I'm currently not showing the year on the page, thus I didn't put it in the list of "in-text fields". Should I add it to that list at the end, so in the screenshot that last one would get a (13)? Or perhaps, because the user wouldn't be able to find a 13 superscript anywhere on the page, it's actually better that it has no number.

**Comment by hdashnow** [comment]
Very nice! Clickable links are working nicely and ordering is much improved.

"2" is assigned to two difference references in the example below. Also, tooltips doesn't work on the references without titles (stuff that failed manubot). Could use the URL name as the title, or I could set a default title in the run-manubot.py script for stuff that fails. In this case I'd just use the database name and ID.

https://deploy-preview-110--strchive.netlify.app/loci/hd_htt/
Screenshot 2024-12-04 at 5 43 04 AM

**Comment by hdashnow** [comment]
For the year, I'd suggest either adding it under the description as below or making that field numerical only and moving the citations to the description section.

image

**Pull request merged**